packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -69,9 +69,8 @@ processors:
   - date:
       field: _tmp.timestamp
       formats:
-        - MMM  d HH:mm:ss
-        - MMM dd HH:mm:ss
-        - MMM d HH:mm:ss
+        - MMM [ ]d HH:mm:ss[.SSSSSS]
+        - MMM [ ]d HH:mm:ss[.SSS]
```

Can we not combine this way?

```suggestion
        - MMM [ ]d HH:mm:ss[.SSS][.SSSSSS]
```

---

i'm trying it now

---

But this would allow `.123.456789` as well which would be weird

---

Seems to work if you reverse it (tested with pipeline simulator):

```
[.SSSSSS][.SSS]
```

---

@ilyannn 
> But this would allow `.123.456789` as well which would be weird

That does not appear to be the case:

```json
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "unable to parse date [Jan 9 15:23:20.123456.987]"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "unable to parse date [Jan 9 15:23:20.123456.987]",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "failed to parse date field [Jan 9 15:23:20.123456.987] with format [MMM [ ]d HH:mm:ss[.SSSSSS][.SSS]]",
      "caused_by": {
        "type": "date_time_parse_exception",
        "reason": "Text 'Jan 9 15:23:20.123456.987' could not be parsed, unparsed text found at index 21"
      }
    }
  }
}
```

My guess is the datetime formatter doesn't like using the same pattern again. Even if it did, I wouldn't be concerned. It would grab either one, but it would be a bogus timestamp anyway, since I'm not aware of any timestamps that do this sort of thing. If for some reason the two numbers had different meanings, I would be more concerned, but that's not the case here.

So to summarize, it will work with either format (3 or 6 digits), but not both at the same time.